src/settings/NoticePolicy/components/DetailSections/components/NoticeCard/NoticeCard.test.js

```diff
@@ -0,0 +1,287 @@
+      );
+      testKeyValues(
+        'noticeCardTriggeringEvent',
+        'triggeringEvent',
+        noticeCardPropsCheckout.triggeringEvents[1].label
+      );
+
+      describe('when "sendOptions" is available', () => {
+        testKeyValues('noticeSend', 'send');
+
+        describe('when "isBeforeOrAfter" is "true"', () => {
+          testKeyValues('noticeCardByText', 'by');
+        });
+
+        describe('when "isFrequencyAvailable" is "true"', () => {
+          testKeyValues('noticeCardFrequencyLabel', 'frequency');
+        });
+
+        describe('when "isRecurring" is "true"', () => {
+          testKeyValues('noticeIsRecurring', 'andEvery');
+        });
+      });
+
+      describe('when "isLoanDueDateTimeSelected" is "true" and real time message set to short term', () => {
+        testKeyValues('isLoanDueDateTimeSelected', 'send.longTerm');
+      });
+
+      describe('when "messageBanner" with warning is shown', () => {
+        testKeyValues('messageBanner', 'postponed.notification');
+      });
+    });
+  });
+
+  describe('with alternative props when "isSendOptionsAvailable" is "false"', () => {
+    const alternativeProps = {
+      ...noticeCardPropsCheckout,
+      notice: {
+        ...noticeCardPropsCheckout.notice,
+        realTime: true,
+        sendOptions: {
+          ...noticeCardPropsCheckout.notice.sendOptions,
+          isSendOptionsAvailable: jest.fn(() => false),
+        },
+      },
+    };
+
+    beforeEach(() => {
+      render(<NoticeCard {...alternativeProps} />);
+    });
+
+    describe('when "sendOptions" is not available', () => {
+      testNotShownValues('noticeSend');
+    });
+
+    describe('when "isLoanDueDateTimeSelected" is "true" and real time message set to short term', () => {
+      testKeyValues('isLoanDueDateTimeSelected', 'send.shortTerm');
+    });
+  });
+
+  describe('with alternative props when "sendOptions" set as "false"', () => {
+    const alternativeProps = {
+      ...noticeCardPropsCheckout,
+      notice: {
+        ...noticeCardPropsCheckout.notice,
+        sendOptions: {
+          ...noticeCardPropsCheckout.notice.sendOptions,
+          isBeforeOrAfter: jest.fn(() => false),
+          isFrequencyAvailable: jest.fn(() => false),
+          isLoanDueDateTimeSelected: jest.fn(() => false),
+        },
+      },
+    };
+
+    beforeEach(() => {
+      render(<NoticeCard {...alternativeProps} />);
+    });
+
+    describe('when "isBeforeOrAfter" is "false"', () => {
+      testNotShownValues('noticeCardByText');
+    });
+
+    describe('when "isFrequencyAvailable" is "false"', () => {
+      testNotShownValues('noticeCardFrequencyLabel');
+    });
+
+    describe('when "isLoanDueDateTimeSelected" is "false"', () => {
+      it('should not render real time message', () => {
+        expect(screen.queryByText('ui-circulation.settings.noticePolicy.notices.send.shortTerm'))
```

Comment:
i think this one and the next translation ids can be moved into `noticeCard` labels map

Reply:
Fixed 
moved to const
const POLICY_NOTICES_TRANSLATION_KEY = 'ui-circulation.settings.noticePolicy.notices';